File: GameLibrary/src/cc/lib/swing/AWTDungeonBuilder.java
package cc.lib.swing;

import java.util.List;

import cc.lib.dungeondice.CellType;
import cc.lib.dungeondice.DBoard;
import cc.lib.dungeondice.DCell;
import cc.lib.game.Utils;

public class AWTDungeonBuilder extends AWTBoardBuilder<DBoard> {

    public static void main(String [] args) {
        new AWTDungeonBuilder();
    }

    @Override
    protected DBoard newBoard() {
        return new DBoard();
    }


    CellType cellType = CellType.EMPTY;

    @Override
    protected void initFrame(AWTFrame frame) {
        super.initFrame(frame);
        frame.addMenuBarMenu("CELL", Utils.toStringArray(CellType.values()));
    }

    @Override
    protected void init(AWTGraphics g) {
        super.init(g);
        cellType = CellType.valueOf(frame.getStringProperty("cellType", cellType.name()));
    }

    @Override
    protected void onMenuItemSelected(String menu, String subMenu) {
        switch (menu) {
            case "CELL":
                onCellMenu(subMenu);
                break;

            default:
                super.onMenuItemSelected(menu, subMenu);
        }
    }

    void onCellMenu(String item) {
        cellType = CellType.valueOf(item);
        frame.setProperty("cellType", cellType.name());
        repaint();
    }

    @Override
    protected void pickCell() {
        super.pickCell();
        if (getSelectedIndex() >= 0) {
            DCell cell = board.getCell(getSelectedIndex());
            cell.setType(cellType);
        }
    }

    @Override
    protected String getPropertiesFileName() {
        return "ddbuilder.properties";
    }

    @Override
    protected String getDefaultBoardFileName() {
        return "ddungeon.backup.board";
    }

    @Override
    protected void getDisplayData(List<String> lines) {
        super.getDisplayData(lines);
        lines.add(cellType.name());
        if (highlightedIndex >= 0) {
            switch (pickMode) {
                case CELL:
                    lines.add(((DCell)board.getCell(highlightedIndex)).getCellType().name());
            }
        }
    }
}

File: SwingApps/apps/cc/applets/dungeonbb/AWTDungeonBuilder.kt
package cc.lib.swing

import cc.lib.board.BEdge
import cc.lib.board.BVertex
import cc.lib.dungeondice.CellType
import cc.lib.dungeondice.DBoard
import cc.lib.dungeondice.DCell
import cc.lib.game.Utils

class AWTDungeonBuilder : AWTBoardBuilder<BVertex, BEdge, DCell, DBoard>() {
	override fun newBoard(): DBoard {
		return DBoard()
	}

	var cellType = CellType.EMPTY
	override fun registerTools() {
		super.registerTools()
		frame.addMenuBarMenu("CELL", *Utils.toStringArray(CellType.values()))
	}

	override fun init(g: AWTGraphics) {
		super.init(g)
		cellType = CellType.valueOf(frame.getStringProperty("cellType", cellType.name))
	}

	override fun onMenuItemSelected(menu: String, subMenu: String) {
		when (menu) {
			"CELL" -> onCellMenu(subMenu)
			else -> super.onMenuItemSelected(menu, subMenu)
		}
	}

	fun onCellMenu(item: String?) {
		cellType = CellType.valueOf(item!!)
		frame.setProperty("cellType", cellType.name)
		repaint()
	}

	/*
    @Override
    protected void pickCellSingleSelect() {
        super.pickCellSingleSelect();
        if (getSelectedIndex() >= 0) {
            DCell cell = board.getCell(getSelectedIndex());
            cell.setType(cellType);
        }
    }
*/
	override val propertiesFileName: String = "ddbuilder.properties"

	override val defaultBoardFileName: String = "ddungeon.backup.board"

	override fun getDisplayData(lines: MutableList<String>) {
		super.getDisplayData(lines)
		lines.add(cellType.name)
		if (highlightedIndex >= 0) {
			when (pickMode) {
				PickMode.CELL -> lines.add((board.getCell(highlightedIndex) as DCell).cellType.name)
				else -> Unit
			}
		}
	}

	companion object {
		@JvmStatic
		fun main(args: Array<String>) {
			AWTDungeonBuilder()
		}
	}
}